plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/button/frontend.ts

```diff
@@ -77,9 +87,30 @@ const productButtonStore = {
 				? tempQuantity || 0
 				: state.quantity;
 
-			if ( quantity === 0 ) return addToCartText;
+			if ( productType === 'grouped' ) {
+				const groupedProductIdsInCart = groupedProductIds?.map(
+					( productId ) => {
+						const product = wooState.cart?.items.find(
+							( item ) => item.id === productId
+						);
+						return product?.quantity || 0;
+					}
+				);
+				if ( groupedProductIdsInCart?.some( ( qty ) => qty > 0 ) ) {
+					return hasPressedButton
+						? state.inTheCartTextGrouped
+						: state.inTheCartText;
+				}
```

Comment:
In grouped products `inTheCartText` is going to have the same text as `addToCartText`, no? So I wonder if we could use that one instead.

If we do that, I think we could also get rid of `inTheCartTextGrouped` altogether and instead use `inTheCartText` to hold the `Added to cart` string (that would require changes in `ProductButton.php`).


```suggestion
				if (
					groupedProductIdsInCart?.some( ( qty ) => qty > 0 ) &&
					hasPressedButton
				) {
					return state.inTheCartText;
				} else {
					return addToCartText;
				}
```

Not a big deal, but I think it will help simplify the code because we will have one less variable to care about. Let me know if it doesn't make sense. :sweat_smile:

Reply:
Yes, that works well, thanks! I was seeing 'Add to cart' persistently on the client-side, so I think that's why I created a new variable just for grouped products. But this is working well, thank you. Done in https://github.com/woocommerce/woocommerce/pull/58002/commits/d1122b0ac923ec0ed264f6359fd8b44f691309fa.